components/server/src/prebuilds/prebuild-manager.ts

```diff
@@ -165,17 +165,21 @@ export class PrebuildManager {
 
         const project = await this.projectService.getProject(user.id, projectId);
 
-        const branchDetails = !!branchName
-            ? await this.projectService.getBranchDetails(user, project, branchName)
-            : (await this.projectService.getBranchDetails(user, project)).filter((b) => b.isDefault);
-        if (branchDetails.length !== 1) {
-            log.debug({ userId: user.id }, "Cannot find branch details.", { project, branchName });
-            throw new ApplicationError(
-                ErrorCodes.NOT_FOUND,
-                `Could not find ${!branchName ? "a default branch" : `branch '${branchName}'`} in repository ${
-                    project.cloneUrl
-                }`,
-            );
+        let branchDetails: Project.BranchDetails[] = [];
+        try {
```

Comment:
This try catch because we used to swallow the branch not found error and when a branch was not found, `projectService.getBranchDetails` has thrown instead of returning empty.

Reply:
`throw` in `finally` will omit the catched error, so any error from getBranchDetails will be omitted. Not sure if it's worth.

Checked `*-repository-provider` files, they will throw Error or ApplicationError (github one, but the code should not hit)

Reply:
Let log the error out with log.error? inside catch (then throw it)

Reply:
<img width="945" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/80c5f5e7-5e5f-4acc-8c78-3c8e0be482ec">

Reply:
Yeah, sorry for misunderstand, we don't need to throw it, just log the error

Reply:
Screenshot is about to test error from finally has higher priority

Reply:
Thank you, I had no idea that throwing in the catch would prefer the error from finally. TIL!